src/Engines/QueryBuilderEngine.php

```diff
@@ -99,8 +99,10 @@ public function count()
      */
     public function filtering()
     {
+        $eagerLoads = array_keys($this->query->getEagerLoads());
```

Comment:
This only works Model but will fail on Query Builder. Maybe we should override filtering on EloquentEngine instead of QueryBuilderEngine.